nucliadb/src/nucliadb/ingest/orm/knowledgebox.py

```diff
@@ -80,78 +85,140 @@ def new_unique_kbid() -> str:
     @processor_observer.wrap({"type": "create_kb"})
     async def create(
         cls,
-        txn: Transaction,
+        driver: Driver,
+        *,
+        kbid: str,
         slug: str,
-        semantic_model: SemanticModelMetadata,
-        uuid: Optional[str] = None,
-        config: Optional[KnowledgeBoxConfig] = None,
+        title: str = "",
+        description: str = "",
+        semantic_model: Optional[SemanticModelMetadata] = None,
+        semantic_models: Optional[dict[str, SemanticModelMetadata]] = None,
         release_channel: Optional[ReleaseChannel.ValueType] = ReleaseChannel.STABLE,
-    ) -> str:
-        release_channel = cast(ReleaseChannel.ValueType, release_channel_for_kb(slug, release_channel))
-
-        exists = await datamanagers.kb.get_kb_uuid(txn, slug=slug)
-        if exists:
-            raise KnowledgeBoxConflict()
-        if uuid is None or uuid == "":
-            uuid = str(uuid4())
+    ) -> tuple[str, str]:
+        """Creates a new knowledge box and return its id and slug."""
 
-        if slug == "":
-            slug = uuid
-
-        await txn.set(
-            datamanagers.kb.KB_SLUGS.format(slug=slug),
-            uuid.encode(),
-        )
-        if config is None:
-            config = KnowledgeBoxConfig()
+        if (semantic_model and semantic_models) or (not semantic_model and not semantic_models):
+            raise KnowledgeBoxCreationError(
+                "KB must only define semantic_model or semantic_models, not both"
+            )
+        if semantic_model is None:
+            if semantic_models is None or len(semantic_models) == 0:
+                raise KnowledgeBoxCreationError(
+                    "KB must be created with at least one semantic model spec"
+                )
+        else:
+            if semantic_models is not None and len(semantic_models) > 0:
+                raise KnowledgeBoxCreationError(
+                    "KB must be created with at least one semantic model spec"
```

Comment:
wrong error message: shoud be something like "KB must only define semantic_model or semantic_models, not both". But aren't you checking the same thing twice?

Reply:
Yes, I left an unnecessary check. Cleaning it up